storageClient/api.go

```diff
@@ -29,6 +29,16 @@ type Note struct {
 	Expires             bool              `json:"expires,omitempty"`
 }
 
+//Group wraps values related to managing client membership and capabilities for resources such as records.
+type Group struct {
```

Comment:
You should separate the difference between a GroupRequest and a GroupResponse -- when we make the request we don't want to have to provide GroupID/CreatedAt/LastModified/AccountID, but we want all those fields when the result is returned to us